src/MCPServer/lib/settings/common.py

```diff
@@ -16,32 +16,176 @@
+    SERVERCONF_CONFIG = """[MCPServer]
+MCPArchivematicaServer = localhost:4730
+watchDirectoryPath = /var/archivematica/sharedDirectory/watchedDirectories/
+sharedDirectory = /var/archivematica/sharedDirectory/
+processingDirectory = /var/archivematica/sharedDirectory/currentlyProcessing/
+rejectedDirectory = %%sharedPath%%rejected/
+watchDirectoriesPollInterval = 1
+processingXMLFile = processingMCP.xml
+waitOnAutoApprove = 0
+
+[Protocol]
+delimiter = <!&\delimiter/&!>
+limitGearmanConnections = 10000
+limitTaskThreads = 75
+limitTaskThreadsSleep = 0.2
+reservedAsTaskProcessingThreads = 8
+    """
+
+    """
+    Mapping configuration attributes with INI section, options and types.
+    """
+    MAPPING = {
+        # [MCPServer]
+        'shared_directory': {'section': 'MCPServer', 'option': 'sharedDirectory', 'type': 'string'},
+        'processing_xml_file': {'section': 'MCPServer', 'option': 'processingXMLFile', 'type': 'string'},
+        'gearman_server': {'section': 'MCPServer', 'option': 'MCPArchivematicaServer', 'type': 'string'},
+        'watch_directory': {'section': 'MCPServer', 'option': 'watchDirectoryPath', 'type': 'string'},
+        'processing_directory': {'section': 'MCPServer', 'option': 'processingDirectory', 'type': 'string'},
+        'rejected_directory': {'section': 'MCPServer', 'option': 'rejectedDirectory', 'type': 'string'},
+        'wait_on_auto_approve': {'section': 'MCPServer', 'option': 'waitOnAutoApprove', 'type': 'int'},
+        'watch_directory_interval': {'section': 'MCPServer', 'option': 'watchDirectoriesPollInterval', 'type': 'int'},
+        'secret_key': {'section': 'MCPServer', 'option': 'django_secret_key', 'type': 'string'},
+
+        # [Protocol]
+        'limit_task_threads': {'section': 'Protocol', 'option': 'limitTaskThreads', 'type': 'int'},
+        'limit_task_threads_sleep': {'section': 'Protocol', 'option': 'limitTaskThreadsSleep', 'type': 'float'},
+        'limit_gearman_conns': {'section': 'Protocol', 'option': 'limitGearmanConnections', 'type': 'int'},
+        'reserved_as_task_processing_threads': {'section': 'Protocol', 'option': 'reservedAsTaskProcessingThreads', 'type': 'int'},
+
+        # [client]
+        'db_engine': {'section': 'client', 'option': 'engine', 'type': 'string'},
+        'db_name': {'section': 'client', 'option': 'database', 'type': 'string'},
+        'db_user': {'section': 'client', 'option': 'user', 'type': 'string'},
+        'db_password': {'section': 'client', 'option': 'password', 'type': 'string'},
+        'db_host': {'section': 'client', 'option': 'host', 'type': 'string'},
+        'db_port': {'section': 'client', 'option': 'port', 'type': 'string'},
+        'db_pool_max_overflow': {'section': 'client', 'option': 'max_overflow', 'type': 'string'},
+    }
+
+    def __init__(self):
+        self.config = EnvConfigParser(prefix=self.ENV_PREFIX)
+
+        self.read_defaults()
+        self.read_files()
```

Comment:
It should be the opposite because `EnvConfigParser` looks up the environment first. If the environment string is not defined the it's going to use the underlying ConfigParser. I wrote some tests here that may help to clarify that: [`test_env_configparser.py`](https://github.com/artefactual/archivematica/blob/stable/1.6.x/src/archivematicaCommon/tests/test_env_configparser.py).

The reason why I used`self.config.readfp` in `read_defaults` and `self.config.read` in `read_files` is because that's how the [docs of ConfigParser](https://docs.python.org/2/library/configparser.html#ConfigParser.RawConfigParser.read) recommend to do it:

> An application which requires initial values to be loaded from a file should load the required file or files using readfp() before calling read() for any optional files.